packages/rendermime/src/widgets.ts

```diff
@@ -162,34 +177,36 @@ class RenderedMarkdown extends RenderedHTMLCommon {
         content = options.sanitizer.sanitize(content);
       }
       Private.appendHtml(this.node, content);
-      if (options.resolver) {
-        this._urlResolved = Private.handleUrls(this.node, options.resolver,
-                                               options.linkHandler);
-      }
       Private.headerAnchors(this.node);
       this.fit();
-      this._rendered = true;
-      if (this.isAttached) {
-        if (this._urlResolved) {
-          this._urlResolved.then(() => { typeset(this.node); });
-        } else {
-          typeset(this.node);
-        }
+      if (options.resolver) {
+        Private.handleUrls(
+          this.node, options.resolver, options.linkHandler).then(() => {
+            this._ready.resolve(undefined);
+        });
+      } else {
+        this._ready.resolve(undefined);
       }
     });
   }
 
   /**
+   * A promise that resolves when the rendered content is ready.
+   */
+  get ready(): Promise<void> {
+    return this._ready.promise;
+  }
+
+  /**
    * A message handler invoked on an `'after-attach'` message.
    */
   onAfterAttach(msg: Message): void {
-    if (this._rendered) {
+    this.ready.then(() => {
```

Comment:
I am misunderstanding the purpose of ready? I was thinking it meant (this is on the page and fully rendered). Based on that I would expect it to typeset and then declare it was ready.